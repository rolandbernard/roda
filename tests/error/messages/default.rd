// test: Test style of error message
// build: ./build/$BUILD/bin/rodac $(realpath --relative-to=. %) 2> %.out || true

extern fn test(a: f64): int;

pub fn foo(a: int): int {
    return test(a);
}
