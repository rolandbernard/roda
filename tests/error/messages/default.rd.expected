error[E0008]: type error, conflicting types `int` and `f64`
    --> tests/error/messages/default.rd:7:17
   3 | 
   4 | extern fn test(a: f64): int;
     |                   --- note: expecting `f64` because of this
   5 | 
   6 | pub fn foo(a: int): int {
     |               --- note: expecting `int` because of this
   7 |     return test(a);
     |                 ^ conflicting types `int` and `f64` for this expression
   8 | }
